### ISSUE TITLE
CookieManager.instance.flush()

### ISSUE THREAD

**Issue by EArminjon** [pichillilorenzo/flutter_inappwebview#2354]
### Is there an existing feature request for this?

- [X] I have searched the existing feature request

### Operating System

Android

### Pain

I've the same issue as https://github.com/react-native-webview/react-native-webview/issues/1345.

If you remove a cookie in the web view and then kill the application in the recent application list on Android. When you launch again the application, the cookie is still here.
He does the same if you set a cookie and kill the app few seconds after, the cookie will not be here.
It seems ok on iOS.

### Suggested solution

Adding a method to manually call Android `getCookieManager().flush();`

https://github.com/react-native-cookies/cookies/blob/69fe6557b0f2c286e6df6fa9100391ad2d163860/android/src/main/java/com/reactnativecommunity/cookies/CookieManagerModule.java#L96

A temporary solution is to delete a random fake cookie : 

```dart
await CookieManager.instance().deleteCookie(url: WebUri('https://example.com'), name: 'i_need_to_call_flush');
```

This will call on Android `getCookieManager().flush();` without regarding if deletion succeed or failed. 

### Useful resources

https://developer.android.com/reference/android/webkit/CookieManager#flush()

### Additional information

Flutter 3.24

### Self grab

- [X] I'm ready to work on this issue!

**Comment by pichillilorenzo** [comment]
flush is already called when setting or deleting cookies on Android, for example:
https://github.com/pichillilorenzo/flutter_inappwebview/blob/f67ae1f34868c7660c16a2473f53a7a7bfb6f784/flutter_inappwebview_android/android/src/main/java/com/pichillilorenzo/flutter_inappwebview_android/MyCookieManager.java#L363

Maybe the flush() method call should be moved inside the `public void onReceiveValue(Boolean successful) { ` callback instead after the `setCookie` method call ??

**Comment by EArminjon** [comment]
Sorry, i was not enough precise.

The issue is present when a cookie is set or removed by a webpage (not by Flutter). In this case, if user kill the app few times after and relaunch it, cookie not be here (in case of creation) or still be here (in case of deletion).

For example we have a scenario where when an users launch the app, they can see a webview to accept or not cookie "consent" (for tracking purpose). Once accepted or rejected we receive a postMessage from our webview to inform us to close the webview. 
We detected this issue in this case : later, if user kill the app and relaunch it, consent is not persisted. 

That's why we found calling .flush() before closing webview seems to well fix the issue.

**Comment by pichillilorenzo** [comment]
Ok, clear now, thanks! So then I will expose that method on Android as soon as possible 👍

**Comment by EArminjon** [comment]
You rock ! Ty so much !

**Comment by EArminjon** [comment]
Exposing flush() will be great and can help other future scenario on Android.

We got also the issue on iOS but this time we didn't have any suggestion / solution. I can create a dedicated issue if you want.

**Comment by pichillilorenzo** [comment]
iOS doesn't have any "flush()" method: https://developer.apple.com/documentation/webkit/wkhttpcookiestore

**Comment by EArminjon** [comment]
Yes, will create new issue for iOS, i will be clean.

**Comment by pichillilorenzo** [comment]
Released new version `6.2.0-beta.1`.
